### ISSUE TITLE
Confusing API Route Interface (even docs are confused)

### ISSUE THREAD

**Issue by dkershner6** [serverless-stack/serverless-stack#502]
Hi there, I ran into some interesting issues while trying to add authorizers to my API.

### Docs

Here is what the docs suggest to add to a specific route:
```typescript
import { HttpJwtAuthorizer } from "@aws-cdk/aws-apigatewayv2-authorizers";

new Api(this, "Api", {
  defaultAuthorizer: new HttpJwtAuthorizer({
    jwtAudience: ["UsGRQJJz5sDfPQDs6bhQ9Oc3hNISuVif"],
    jwtIssuer: "https://myorg.us.auth0.com",
  }),
  routes: {
    "GET /public": "src/public.main",
    "GET /private": {
      function: {
        handler: "src/private.main",
        authorizationType: ApiAuthorizationType.JWT,
      },
    },
  },
});
```

You will note that the `authorizationType` is within the `function` object.  This is improper, as it should be outside it.  This occurs twice in the docs.

### Confusing Interface

Because the interface for a route accepts both a `FunctionDefinition` OR a `FunctionDefinition` in a `function` object, you can get interesting interactions.

For example, this will not throw a type error: 
```typescript
"GET /private": {
        handler: "src/private.main",
        authorizationType: ApiAuthorizationType.JWT,
    },
```

It will throw this error on synth, however:

```
TypeError: Cannot read property 'handler' of undefined
    at Function.fromDefinition (/home/dkershner/repos/global-api-devops/node_modules/@serverless-stack/resources/src/Function.ts:328:18)
    at Api.addRoute (/home/dkershner/repos/global-api-devops/node_modules/@serverless-stack/resources/src/Api.ts:339:23)
    at /home/dkershner/repos/global-api-devops/node_modules/@serverless-stack/resources/src/Api.ts:219:23
    at Array.forEach (<anonymous>)
    at Api.addRoutes (/home/dkershner/repos/global-api-devops/node_modules/@serverless-stack/resources/src/Api.ts:217:25)
    at DevOpsApiStack.createApi (/home/dkershner/repos/global-api-devops/lib/DevOpsApiStack.ts:194:13)
    at new DevOpsApiStack (/home/dkershner/repos/global-api-devops/lib/DevOpsApiStack.ts:34:14)
    at Object.main (/home/dkershner/repos/global-api-devops/lib/index.ts:22:5)
    at Object.<anonymous> (/home/dkershner/repos/global-api-devops/.build/run.js:77:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
```

I think my preference would be to force all definitions to use the `function` object, but I will leave that decision to the group.

**Comment by fwang** [comment]
Hey @dkershner6, thanks for pointing it out. It is indeed confusing.. we got it mixed up in the doc ourselves 😞 (just fixed it)

One of our design principle is to make the minimal version (for all constructs) as easy to understand as possible (for newcomers),
```js
new Api(this, "Api", {
  routes: {
    "GET /public": "src/public.main",
    "GET /private": "src/private.main",
  },
});
```
while having the ability for complete customization.

That said, "confusing interface" is a serious issue. I just bumped up the priority on #486 to validate construct interface. And in this case, ideally we want to show an error message that says something like:
```
authorizationType is not a property of FunctionDefinition. Did you mean to configure it on the route instead?
```
Let me put the validation in. And if ppl still get confused, we might need tweak the interface like you suggested.
